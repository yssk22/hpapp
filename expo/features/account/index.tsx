import ListItemClearCache from './internal/ListItemClearCache';

export { ListItemClearCache };
