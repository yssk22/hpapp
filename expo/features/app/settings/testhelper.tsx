import { act, render } from '@testing-library/react-native';

import { SettingsProvider, SettingsProviderProps } from './';
import SettingsList from './internals/SettingsList';

/**
 * Clear all settings
 */
export async function clearSettings() {
  await Promise.all(
    SettingsList.map((settings) => {
      settings.clear();
    })
  );
}

export async function renderSettingsProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  await clearSettings();
  const content = render(<SettingsProvider {...props}>{children}</SettingsProvider>);
  await act(async () => {});
  return content;
}
