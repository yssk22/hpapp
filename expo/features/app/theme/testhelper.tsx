import { SettingsProviderProps } from '@hpapp/features/app/settings';
import { renderSettingsProvider } from '@hpapp/features/app/settings/testhelper';

import { ThemeProvider } from './';

export async function renderThemeProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  return renderSettingsProvider(<ThemeProvider>{children}</ThemeProvider>, props);
}
