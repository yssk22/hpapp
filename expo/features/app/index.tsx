import { maybeDev } from '@hpapp/foundation/environment';
import { LocalMediaManagerProvider } from '@hpapp/system/media';

import AppRelayProvider from './internals/AppRelayProvider';
import { default as Root, AppRootProps as RootProps } from './internals/AppRoot';
import AppUpdateBanner from './internals/AppUpdateBanner';
import { ThemeProvider, ThemeProviderProps } from './theme';

export type AppRootProps = RootProps & ThemeProviderProps;

export function AppRoot({ screens, ...props }: AppRootProps) {
  return (
    <ThemeProvider {...props}>
      <AppRelayProvider>
        <LocalMediaManagerProvider name={maybeDev() ? 'hellofanapp-dev' : 'hellofanapp'}>
          <AppUpdateBanner />
          <Root screens={screens} />
        </LocalMediaManagerProvider>
      </AppRelayProvider>
    </ThemeProvider>
  );
}
