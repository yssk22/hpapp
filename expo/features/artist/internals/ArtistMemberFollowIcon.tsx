import { useThemeColor } from '@hpapp/features/app/theme';
import { HPFollowType, HPMember, useHelloProject, useMe } from '@hpapp/features/app/user';
import { ThemeColorScheme } from '@hpapp/system/theme';
import { Icon } from '@rneui/themed';

function getIconName(type: HPFollowType) {
  switch (type) {
    case 'follow':
      return 'account-check';
    case 'follow_with_notification':
      return 'bell-check';
    default:
      return null;
  }
}

export default function ArtistMemberFollowIcon({
  colorScheme = 'primary',
  member,
  size
}: {
  member: HPMember | string;
  colorScheme?: ThemeColorScheme;
  size?: number;
}) {
  const hp = useHelloProject();
  const me = useMe();
  const m = hp.useMember(member);
  const followType = me.useFollowType(m!.id);
  const iconName = getIconName(followType);
  const [color] = useThemeColor(colorScheme);
  if (iconName === null) {
    return null;
  }

  return <Icon testID="ArtistMemberFollowIcon" type="material-community" name={iconName} size={size} color={color} />;
}
