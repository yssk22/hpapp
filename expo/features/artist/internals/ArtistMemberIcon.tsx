import { useThemeColor } from '@hpapp/features/app/theme';
import { HPFollowType, HPMember, useHelloProject, useMe } from '@hpapp/features/app/user';
import { ExternalImage } from '@hpapp/features/common';
import { IconSize } from '@hpapp/features/common/constants';
import { Icon } from '@rneui/themed';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

export enum ArtistMemberIconSize {
  // eslint-disable-next-line no-unused-vars
  Small = 32,
  // eslint-disable-next-line no-unused-vars
  Medium = 64,
  // eslint-disable-next-line no-unused-vars
  Large = 96
}

export type ArtistMemberIconProps = {
  member: HPMember | string;
  size?: ArtistMemberIconSize | number;
  circle?: boolean;
  showFollowIcon?: boolean;
  onPress?: () => void;
};

export default function ArtistMemberIcon({
  member,
  size = ArtistMemberIconSize.Medium,
  circle = false,
  showFollowIcon = false,
  onPress
}: ArtistMemberIconProps) {
  const [color, contrast] = useThemeColor('secondary');
  const hp = useHelloProject();
  const me = useMe();
  const m = hp.useMember(member);
  const followType = me.useFollowType(m!.id);
  const iconName = getFollowIconName(followType);
  const circleProps = circle ? { borderRadius: size * 2 } : {};
  if (onPress === undefined) {
    return (
      <View
        testID="ArtistMemberIcon.NoOnPress"
        style={{
          width: size,
          height: size
        }}
      >
        <ExternalImage
          uri={m!.thumbnailURL}
          style={{ width: size, height: size }}
          width={size}
          height={size}
          cachePolicy="disk"
          {...circleProps}
        />
        {showFollowIcon && iconName && (
          <View style={[{ backgroundColor: contrast }, styles.icon]}>
            <Icon name={iconName} type="material-community" color={color} size={IconSize.Small} />
          </View>
        )}
      </View>
    );
  }
  return (
    <TouchableOpacity
      testID="ArtistMemberIcon"
      style={{
        width: size,
        height: size
      }}
      onPress={onPress}
    >
      <ExternalImage
        uri={m!.thumbnailURL}
        style={{ width: size, height: size }}
        width={size}
        height={size}
        cachePolicy="disk"
        {...circleProps}
      />
      {showFollowIcon && iconName && (
        <View style={[{ backgroundColor: contrast }, styles.icon]}>
          <Icon name={iconName} type="material-community" color={color} size={IconSize.Small} />
        </View>
      )}
    </TouchableOpacity>
  );
}

function getFollowIconName(type: HPFollowType) {
  switch (type) {
    case 'follow':
      return 'account-check';
    case 'follow_with_notification':
      return 'bell-check';
    default:
      return null;
  }
}

const styles = StyleSheet.create({
  icon: {
    position: 'absolute',
    right: 0, // IconSize.Small / 2 - 1,
    top: 0, // Spacing.Small,
    alignItems: 'flex-end'
  }
});
