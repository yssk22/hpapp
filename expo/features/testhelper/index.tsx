import {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered
} from './internals/jest';
import { ScreenDecorator } from './internals/storybook';

export {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered,
  ScreenDecorator
};
