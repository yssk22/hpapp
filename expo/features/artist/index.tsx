import ArtistMemberFollowIcon from './internals/ArtistMemberFollowIcon';
import ArtistMemberIcon, { ArtistMemberIconSize } from './internals/ArtistMemberIcon';

export { ArtistMemberFollowIcon, ArtistMemberIcon, ArtistMemberIconSize };
