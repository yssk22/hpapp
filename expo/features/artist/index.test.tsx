import { renderUserComponent } from '@hpapp/features/testhelper';

import ArtistMemberFollowIcon from './internals/ArtistMemberFollowIcon';
import ArtistMemberIcon from './internals/ArtistMemberIcon';

describe('artist', () => {
  describe('ArtistMemberFollowIcon', () => {
    test('following', async () => {
      const content = await renderUserComponent(<ArtistMemberFollowIcon member="8589934592" />);
      const icon = content.queryByTestId('ArtistMemberFollowIcon');
      expect(icon).toBeTruthy();
      expect(content).toMatchSnapshot();
    });

    test('not following', async () => {
      const content = await renderUserComponent(<ArtistMemberFollowIcon member="8589934593" />);
      const icon = content.queryByTestId('ArtistMemberFollowIcon');
      expect(icon).toBeFalsy();
      expect(content).toMatchSnapshot();
    });
  });

  describe('ArtistMemberIcon', () => {
    test('icon no press', async () => {
      const content = await renderUserComponent(<ArtistMemberIcon member="8589934592" />);
      const icon = content.queryByTestId('ArtistMemberIcon.NoOnPress');
      expect(icon).toBeTruthy();
      expect(content).toMatchSnapshot();
    });

    test('icon with press', async () => {
      const content = await renderUserComponent(<ArtistMemberIcon member="8589934592" onPress={() => {}} />);
      const icon = content.queryByTestId('ArtistMemberIcon');
      expect(icon).toBeTruthy();
      expect(content).toMatchSnapshot();
    });
  });
});
