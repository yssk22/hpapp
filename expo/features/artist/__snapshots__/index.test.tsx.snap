// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artist ArtistMemberFollowIcon following 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <Icon
              color="#0075c2"
              name="bell-check"
              testID="ArtistMemberFollowIcon"
              type="material-community"
            />
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</View>
`;

exports[`artist ArtistMemberFollowIcon not following 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          />
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</View>
`;

exports[`artist ArtistMemberIcon icon no press 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "height": 64,
                  "width": 64,
                }
              }
              testID="ArtistMemberIcon.NoOnPress"
            >
              <Image
                source={
                  {
                    "uri": "http://cdn.helloproject.com/img/artist/m/8bf17e09fa5cb3e62ad57617c81e1b6c8f9958f6.jpg",
                  }
                }
                style={
                  [
                    {
                      "height": 64,
                      "width": 64,
                    },
                    {
                      "height": 64,
                      "width": 64,
                    },
                  ]
                }
                testID="ExternalImage.Image"
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</View>
`;

exports[`artist ArtistMemberIcon icon with press 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "height": 64,
                  "opacity": 1,
                  "width": 64,
                }
              }
              testID="ArtistMemberIcon"
            >
              <Image
                source={
                  {
                    "uri": "http://cdn.helloproject.com/img/artist/m/8bf17e09fa5cb3e62ad57617c81e1b6c8f9958f6.jpg",
                  }
                }
                style={
                  [
                    {
                      "height": 64,
                      "width": 64,
                    },
                    {
                      "height": 64,
                      "width": 64,
                    },
                  ]
                }
                testID="ExternalImage.Image"
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</View>
`;
